src/board.h

```diff
@@ -0,0 +1,137 @@
+/**
+ * @file board.h
+ * @brief Board state for Numbers Royale
+ */
+#include <vector>
+#include <string>
+#include <memory>
+#include "player.h"
+#include "gui.h"
+
+
+class Board
+{
+private:
+  std::vector<std::unique_ptr<Player>> players_;
+  unsigned int num_humans_;
+  unsigned int board_size_;
+
+public:
+  // Globals
+  static constexpr size_t DEFAULT_BOARD_SIZE = 9;
+  static constexpr int CONTINUE_GAME = -1;
+  static constexpr int TIE_GAME = 0;
+
+  // Comparator functions
+  // Sort by descending score
+  static bool score_cmp(const Player* a, const Player* b) {
+    return a->get_score() > b->get_score();
+  }
+  // Sort by last move, descending
+  static bool move_cmp(const Player* a, const Player* b) {
+    return a->get_last_move() > b->get_last_move();
+  }
```

Comment:
I would just have this in an anon namespace in .cpp file. No need for this to be part of the class, nor public at least

Reply:
it makes more sense for me to have this in the player class, but no strong opinion